applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_cut_pfem_process.hpp

```diff
@@ -0,0 +1,1706 @@
+
+						const array_1d<double, 3> &node0_data = slave_node_1->FastGetSolutionStepValue(variable, step);
+						const array_1d<double, 3> &node1_data = slave_node_2->FastGetSolutionStepValue(variable, step);
+
+						noalias(node_data) = (0.5 * node0_data + 0.5 * node1_data);
+						// node_data = (0.5*node0_data + 0.5*node1_data);
+					}
+				}
+				else if (KratosComponents<Variable<int>>::Has(variable_name))
+				{
+					// std::cout<<"int"<<std::endl;
+					// NO INTERPOLATION
+				}
+				else if (KratosComponents<Variable<bool>>::Has(variable_name))
+				{
+					// std::cout<<"bool"<<std::endl;
+					// NO INTERPOLATION
+				}
+				else if (KratosComponents<Variable<Matrix>>::Has(variable_name))
+				{
+					// std::cout<<"Matrix"<<std::endl;
+					const Variable<Matrix> &variable = KratosComponents<Variable<Matrix>>::Get(variable_name);
+					for (SizeType step = 0; step < buffer_size; step++)
+					{
+						// getting the data of the solution step
+						Matrix &node_data = master_node->FastGetSolutionStepValue(variable, step);
+
+						Matrix &node0_data = slave_node_1->FastGetSolutionStepValue(variable, step);
+						Matrix &node1_data = slave_node_2->FastGetSolutionStepValue(variable, step);
+
+						if (node_data.size1() > 0 && node_data.size2())
+						{
+							if (node_data.size1() == node0_data.size1() && node_data.size2() == node0_data.size2() &&
+								node_data.size1() == node1_data.size1() && node_data.size2() == node1_data.size2())
+							{
+
+								noalias(node_data) = (0.5 * node0_data + 0.5 * node1_data);
+								// node_data = (0.5*node0_data + 0.5*node1_data);
+							}
+						}
+					}
+				}
+				else if (KratosComponents<Variable<Vector>>::Has(variable_name))
+				{
+					// std::cout<<"Vector"<<std::endl;
+					const Variable<Vector> &variable = KratosComponents<Variable<Vector>>::Get(variable_name);
+					for (SizeType step = 0; step < buffer_size; step++)
+					{
+						// getting the data of the solution step
+						Vector &node_data = master_node->FastGetSolutionStepValue(variable, step);
+
+						Vector &node0_data = slave_node_1->FastGetSolutionStepValue(variable, step);
+						Vector &node1_data = slave_node_2->FastGetSolutionStepValue(variable, step);
+
+						if (node_data.size() > 0)
+						{
+							if (node_data.size() == node0_data.size() &&
+								node_data.size() == node1_data.size())
+							{
+
+								noalias(node_data) = (0.5 * node0_data + 0.5 * node1_data);
+								// node_data = (0.5*node0_data + 0.5*node1_data);
+							}
+						}
+					}
+				}
+			}
+
+			KRATOS_CATCH("")
+		}
+
+		void TakeMaterialPropertiesFromNotRigidNode(Node::Pointer master_node, Node::Pointer SlaveNode)
+		{
+
+			KRATOS_TRY
+			master_node->FastGetSolutionStepValue(PROPERTY_ID) = SlaveNode->FastGetSolutionStepValue(PROPERTY_ID);
+			if (master_node->SolutionStepsDataHas(BULK_MODULUS) && SlaveNode->SolutionStepsDataHas(BULK_MODULUS))
```

Comment:
side comment, why not defining a method like:

~~~cpp
void SetMasterCustomVariableFromSlave (Variable<double> rVariable) {
    if (master_node->SolutionStepsDataHas(rVariable) && SlaveNode->SolutionStepsDataHas(rVariable))
    master_node->FastGetSolutionStepValue(rVariable) = SlaveNode->FastGetSolutionStepValue(rVariable);
}

And then apply it to a list of all these variables?
~~~